### PR TITLE
Migrate sds.c unit tests to new test framework.

### PR DESCRIPTION
**Pull request by zhulipeng**
This patch migrates all tests in sds.c into new test framework as part of the parent issue https://github.com/valkey-io/valkey/issues/428.

### CODE REVIEW

**Comment by zhulipeng** [comment]
Didn't bring in the assertion change in sds.c of https://github.com/valkey-io/valkey/pull/476 here.

**Review by karthyuom — APPROVED** [review]
LGTM.

File: src/unit/test_sds.c [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+        sdslen(x) == sizeof(etalon) && memcmp(x, etalon, sizeof(etalon)) == 0);
+
+    sdsfree(x);
+    x = sdsnew("--");
+    x = sdscatfmt(x, "Hello %s World %I,%I--", "Hi!", LLONG_MIN,LLONG_MAX);
+    TEST_ASSERT_MESSAGE("sdscatfmt() seems working in the base case",
+        sdslen(x) == 60 &&
+        memcmp(x,"--Hello Hi! World -9223372036854775808,"
+                    "9223372036854775807--",60) == 0);
+    printf("[%s]\n", x);
```
Comment (madolson) [inline src/unit/test_sds.c]: We don't need all this leftover debugging information.

Comment (madolson) [reply src/unit/test_sds.c]:
```suggestion
```

**Review by madolson — APPROVED** [review]

**Pull request merged**